skimage/_shared/tests/test_kahan.py

```diff
@@ -0,0 +1,31 @@
+from skimage._shared.kahan import kahan_sum, kahan_cumsum
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+
+
+def test_sum():
+    rnd = np.random.RandomState(0)
+    z = rnd.random_sample((50, 50, 3))
+
+    assert_array_almost_equal(kahan_sum(z), np.sum(z))
+    assert_array_almost_equal(kahan_sum(z, axis=1),
+                              np.sum(z, axis=1))
+    assert_array_almost_equal(kahan_sum(z, axis=0),
+                              np.sum(z, axis=0))
+
```

Comment:
is it possible to add a test where kahan summation would result in a better precision than ``np.sum``? I tried with the sum of ``np.arange(n)`` since the result is known analytically, but the result was exact for ``np.sum``.